### ISSUE TITLE
Allow user to request vertical text alignment without manually setting offsets

### ISSUE THREAD

**Issue by heroin-moose** [polybar/polybar#1837]
## Is your feature request related to a problem? Please describe.
Manual offset values are not always ideal and different fonts require different values (e.g. Font Awesome and Fira Mono). It would be much either to just say "font-0 = "Foo Bar Medium 11;center".  

## Why does polybar need this feature?
It would slightly improve user experience.

## Describe the solution you'd like
Additional option that allows user to vertically align text at the center of the polybar.

## Describe alternatives you've considered
Manually settings vertical offsets (as we are doing now).

**Comment by Lomadriel** [comment]
It seems to be a good idea, but this isn't the priority for now.
Now we focus on reducing the number of bugs, when done, I will try to implement this.

**Comment by patrick96** [comment]
It will also be a challenge to even figure out what "center" means for a font programatically

**Comment by heroin-moose** [comment]
@patrick96, why? It seems like there is a code snippet on the Cairo website: https://cairographics.org/samples/text_align_center.

**Comment by Lomadriel** [comment]
Yeah I was thinking about using the origin line and the bearing as said by @heroin-moose.

**Comment by Lomadriel** [comment]
I modified the following line:
https://github.com/polybar/polybar/blob/fe7dc28b07afe949ee2e0fae625e2e0d456d0185/include/cairo/context.hpp#L200

by
```cpp
          f->render(subset, x, y - (extents.height / 2 + extents.y_bearing));
```

And it seems to work pretty well:
![polybar](https://user-images.githubusercontent.com/10365269/61865523-a751d000-aed3-11e9-9347-7615549daa4b.png)

For now this is a temporary solution. I'll open a PR with a real solution.
